src/sentry/tasks/post_process.py

```diff
@@ -595,6 +595,37 @@ def process_rules(job: PostProcessJob) -> None:
     return
 
 
+def process_code_mappings(job: PostProcessJob) -> None:
+    if job["is_reprocessed"]:
+        return
+
+    from sentry.tasks.derive_code_mappings import derive_code_mappings
+
+    try:
+        event = job["event"]
+        project = event.project
+
+        cache_key = f"code-mappings:{project.id}"
+        project_queued = cache.get(cache_key)
+        if project_queued is None:
+            cache.set(cache_key, True, 3600)
+
+        if project_queued or not features.has(
+            "organizations:derive-code-mappings", event.project.organization
+        ):
+            return
+
+        if killswitch_matches_context(
+            "post_process.derive-code-mappings",
+            {"project_id": project.id},
+        ):
+            return
+        derive_code_mappings.delay(project.organization_id, project.id, event.data)
```

Comment:
I _prefer_ only sending `event.data` so all the logic is derived in there (even though I know it feels that repeating code over there; I find it clearer since it saves looking for the code of the caller).

Reply:
I think this is alright since the only additional param is now the project ID. Otherwise, we would need to send the whole event to derive the project since the `project_id` isn't contained in `event.data`.